src/coreclr/debug/createdump/crashreportwriter.cpp

```diff
@@ -84,18 +84,20 @@ CrashReportWriter::WriteCrashReport()
     {
         OpenObject();
         bool crashed = false;
-        if (thread->ManagedExceptionObject() != 0)
+        if (thread->Tid() == m_crashInfo.CrashThread())
         {
             crashed = true;
-            exceptionType = "0x05000000";   // ManagedException
-        }
-        else
-        {
-            if (thread->Tid() == m_crashInfo.CrashThread())
+            if (thread->ManagedExceptionObject() != 0)
```

Comment:
One thing that I was thinking here: In the unwindThread paths we check for LastThrown and request flags on the ExceptionState and do `(flags & Ex_IsUnhandled)`

Reply:
So only use LastThrown if Ex_IsUnhandled is set?

Reply:
Yes

Reply:
But how do I get this flag in createdump. It looks like an internal vm flag

Reply:
The DAC has an API to get the flags on an ExceptionState I thought.